config/environments/test.rb

```diff
@@ -41,12 +41,12 @@
   config.serve_static_assets = true
   config.static_cache_control = "public, max-age=3600"
 
-  # Raise exception on mass assignment protection for Active Record models
-  config.active_record.mass_assignment_sanitizer = :strict
-
   #enables a few aliases - context, should, and should_eventually methods
   config.minitest_spec_rails.mini_shoulda = true
 
   # Use separate cache stores for parallel_tests
   config.cache_store = :file_store, Rails.root.join("tmp", "cache", "paralleltests#{ENV['TEST_ENV_NUMBER']}")
+  config.active_record.mass_assignment_sanitizer = :strict
+
+  config.active_record.maintain_test_schema = false
```

Comment:
Ah sorry, that's the most recent change. Remember how on Rails 4 the test database is auto created (https://github.com/theforeman/foreman/pull/2870#issuecomment-152564961) ? This prevents it. 

Without this, running jenkins:unit would automatically create a database. This happened *after* initializing models with `validate_inclusion_in_families`, so the validation wasn't set up properly because `attribute_names` was empty at that stage (no database). Therefore some tests would fail on the first run because of the autocreated db. On the second and successive runs, with an already existing db they'd work fine because `attribute_names` contains the attributes.

We control in `jenkins.rake` when the database is created, which should be before the environment is loaded. By controlling that we ensure `attribute_names` runs after the db is created and therefore `validate_inclusion_in_famliles` works